js_modules/dagit/packages/core/src/instance/InstanceConcurrency.tsx

```diff
@@ -0,0 +1,530 @@
+import {gql, useQuery, useMutation} from '@apollo/client';
+import {
+  PageHeader,
+  Heading,
+  Box,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Icon,
+  Menu,
+  MenuItem,
+  Mono,
+  Popover,
+  Spinner,
+  ButtonLink,
+  Table,
+  Tag,
+  TextInput,
+  Button,
+} from '@dagster-io/ui';
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+
+import {useFeatureFlags} from '../app/Flags';
+import {useQueryRefreshAtInterval, FIFTEEN_SECONDS} from '../app/QueryRefresh';
+import {useTrackPageView} from '../app/analytics';
+import {useDocumentTitle} from '../hooks/useDocumentTitle';
+import {RunTable, RUN_TABLE_RUN_FRAGMENT} from '../runs/RunTable';
+import {titleForRun} from '../runs/RunUtils';
+
+import {InstancePageContext} from './InstancePageContext';
+import {InstanceTabs} from './InstanceTabs';
+import {
+  InstanceConcurrencyLimitsQuery,
+  InstanceConcurrencyLimitsQueryVariables,
+  ConcurrencyLimitFragment,
+  SetConcurrencyLimitMutation,
+  SetConcurrencyLimitMutationVariables,
+} from './types/InstanceConcurrency.types';
+
+const RUNS_LIMIT = 25;
+
+const InstanceConcurrencyPage = React.memo(() => {
+  useTrackPageView();
+  useDocumentTitle('Configuration');
+  const {flagInstanceConcurrencyLimits} = useFeatureFlags();
+
+  const {pageTitle} = React.useContext(InstancePageContext);
+  const queryResult = useQuery<
+    InstanceConcurrencyLimitsQuery,
+    InstanceConcurrencyLimitsQueryVariables
+  >(INSTANCE_CONCURRENCY_LIMITS_QUERY, {
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const refreshState = useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
+  const {data} = queryResult;
+
+  if (!data) {
+    return (
+      <Box padding={{vertical: 64}}>
+        <Spinner purpose="section" />
+      </Box>
+    );
+  }
+
```

Comment:
Might be nice to show the `PageHeader` and tabs, with the spinner underneath that. That way the user can navigate away to another tab if it's too slow, etc.